### PR TITLE
Multi-factor authentication

### PR DESCRIPTION
**Pull request by alesstimec**

### CODE REVIEW

File: cmd/candidsrv/config.yaml [hunk 1]
```diff
@@ -11,10 +11,14 @@ max-mgo-sessions: 300
...
 private-addr: localhost
 admin-agent-public-key: iKp24EL2Aj9MQfRkpwzp7rz7Zf3QZsEzWGpoWT3OK2w=
+mfa-rp-display-name: Candid
+mfa-rp-id: localhost
+mfa-rp-origin: http://localhost:8081
 identity-providers:
 - type: static
   name: static
   description: Default identity provider
+  require-2fa: true
```
Comment (mhilton) [inline cmd/candidsrv/config.yaml]: could this be `require-mfa` so it's consistent?

File: idp/static/static.go [hunk 2]
```diff
@@ -194,9 +197,11 @@ func (idp *identityProvider) Handle(ctx context.Context, w http.ResponseWriter,
 		id, err := idputil.HandleLoginForm(ctx, w, req, idpChoice, idp.initParams.Template, idp.loginUser)
 		if err != nil {
 			idp.initParams.VisitCompleter.RedirectFailure(ctx, w, req, ls.ReturnTo, ls.State, err)
+			return
 		}
 		if id != nil {
-			idp.initParams.VisitCompleter.RedirectSuccess(ctx, w, req, ls.ReturnTo, ls.State, id)
+			idp.initParams.VisitCompleter.RedirectMFA(ctx, w, req, idp.params.Require2FA, ls.ReturnTo, ls.State, state, id)
```
Comment (mhilton) [inline idp/static/static.go]: This seems an odd way to do this, I assume RedirectMFA falls back to Redirect success if Require2FA is false?

**Review by mhilton — APPROVED** [review]
This can be done in a follow-up, but can we be completely consistent about whether to use "MFA" or "2FA" throughout the code please?

File: idp/static/static.go (hunk 2 above)
Comment (alesstimec) [reply idp/static/static.go]: yes

**Pull request merged**